registrations/api.py

```diff
@@ -383,15 +384,33 @@ def get_serializer_class(self):
             return SignUpGroupCreateSerializer
         return super().get_serializer_class()
 
+    def _ensure_shared_request_data(self, data):
+        if self.action in ("update", "partial_update"):
+            registration_id = (
+                data.get("registration") or self.get_object().registration_id
```

Comment:
is there some particual reason not to use dict's default value? `data.get("registration", self.get_object().registration_id)` would make my pythonic eyes more sense. 
Obivously on creation it will blow up since the `get_object` returns `None` but that has been tackled in the above so should be the case.

Reply:
Here I was thinking the case where `registration` would be an empty string. In that case the usage of `or` would lead to the `self.get_object().registration_id` being used and thus a somewhat unnecessary validation error would be avoided. If it was defined as the dict's default value, it would not be used as the empty string would be considered a usable value.

Thinking about this again I realized that there is going to be a validation error regardless of this as the `self.get_object().registration_id` is only going to be passed to the "children" of the group (sign-ups). The parent/group is going to have the empty string in its data regardless of this code and a validation error will occur.

The code could be easily improved to "fix" also the parent validation error, but perhaps it is best to raise the error and not try to alter the request data for a smoother API experience. So I'll commit this change soon.

Reply:
Changed - will re-request review soon